src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -86,12 +88,75 @@ class SnykLanguageClient :
     private val progressEndMsgCache: Cache<String, WorkDoneProgressEnd> =
         Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS).build()
 
+
     override fun telemetryEvent(`object`: Any?) {
         // do nothing
     }
 
-    override fun publishDiagnostics(diagnostics: PublishDiagnosticsParams?) {
-        // do nothing for now
+    override fun publishDiagnostics(diagnosticsParams: PublishDiagnosticsParams?) {
+        if (diagnosticsParams == null) {
+            return
+        }
+
+        updateCache(diagnosticsParams)
+    }
+
+    private fun updateCache(diagnosticsParams: PublishDiagnosticsParams) {
+        /*
+        If we have diagnostics for file app.js for product code
+            Delete diagnostics for product code file app.js from the cache
+
+        If we have empty diagnostics for file app.js product code
+            Delete diagnostics for product code file app.js from the cache
+
+        If issues exists
+            add new findings to the cache for app.js product code ONLY
+         */
+
+        val filePath = diagnosticsParams.uri
+
+        try {
+            getScanPublishersFor(filePath.toVirtualFile().path).forEach { (project, scanPublisher) ->
+                val snykFile = SnykFile(project, filePath.toVirtualFile())
+                val snykCachedResults = getSnykCachedResults(project) ?: return
```

Comment:
The cache should not be manipulated on the sending side, but on the receiving side.